client/js/widgets/widget.js

```diff
@@ -942,6 +942,24 @@ export class Widget extends StateManaged {
         }
         if((a.property == 'parent' || a.property == 'deck') && a.value !== null && !widgets.has(a.value)) {
           problems.push(`Tried setting ${a.property} to ${a.value} which doesn't exist.`);
+        } else if (a.property == 'id' && isValidCollection(a.collection)) {
+          if (collections[a.collection].length) {
+            const oldWidget = collections[a.collection][0];
+            var newState = Object.assign({},oldWidget.state);
+            const oldState = JSON.stringify(oldWidget.state);
+
+            newState.id = compute(a.relation, null, oldWidget.get(a.property), a.value);
+            if(!widgets.has(newState.id)) {
+              $('#editWidgetJSON').dataset.previousState = oldState;
+              $('#editWidgetJSON').value = JSON.stringify(newState);
+              await onClickUpdateWidget(false);
+              sendDelta(true);
+            } else {
+              problems.push(`New id already in use, ignored.`);
+            }
+          } else {
+            problems.push(`Collection ${a.collection} is empty, ignored.`);
```

Comment:
In my PR, zero length collections are simply ignored, just like when setting other properties.

Reply:
I'll just remove this else, then. No reason for a separate PR for that.

Reply:
Forgot to mention it in my review (and didn't make the change myself) but it would also be nice if problems specified the actual id that failed.

Reply:
Just to be clear: the separate PR would be merged into this one before merging this one. It's just a proposed change to _this PR_.

Reply:
It could possibly be rebased to main, but yes, my intention was that it could just be merged to this PR.